### PR TITLE
[DEVHUB-1546] Consume @mdb/devcenter-components for filter components.

### PR DESCRIPTION
**Pull request by mcolella14**
## Jira Ticket:

[DEVHUB-1546](https://jira.mongodb.org/browse/DEVHUB-1546)

## Description:

Replaces all functionality of the `FilterGroup`, `FilterTag`, and `FIlterTagSection` components with the corresponding components from `@mdb/devcenter-components`. 

Since the shared components had no need for a radio filter input (and since this depends on a feature not yet in the main flora release), I modified the existing `FilterGroup` component to become `RadioFilterGroup`, fulfilling the specific need we have. Once the radio button functionality is available in Flora, this may be added to `@mdb/devcenter-components`.

Everything should functionally and visually be the same in the app.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-    "extends": "next/core-web-vitals"
+    "extends": ["next/core-web-vitals", "plugin:react/jsx-runtime"]
```
Comment (mcolella14) [inline .eslintrc.json:2]: Was having some issues with eslint and parsing the `sx` prop with the Theme UI jsx, this seems to help.

File: src/components/search-filters/desktop/filters.tsx [hunk 2]
```diff
@@ -13,64 +14,88 @@ const DesktopFilters: React.FunctionComponent<FiltersProps> = ({
     contentTypeItems = [],
     codeLevelItems = [],
 }) => {
+    const onToggle = (checked: boolean, filter: FilterItem) => {
+        if (checked) {
+            if (filter.subFilters && filter.subFilters.length) {
```
Comment (justinmckenzie) [inline src/components/search-filters/desktop/filters.tsx:19]: I think you might be able to shorten this to `if (filter?.subFilters.length)`

File: src/components/search-filters/mobile/filters.tsx [hunk 3]
```diff
@@ -33,9 +36,33 @@ const MobileFilters: React.FunctionComponent<MobileFiltersProps> = ({
     closeModal,
 }) => {
     const [tempFilters, setTempFilters] = useState<FilterItem[]>(allFilters);
-    const onTempFilter = (filters: FilterItem[]) => {
-        setTempFilters(filters);
+    const onToggle = (checked: boolean, filter: FilterItem) => {
+        if (checked) {
+            if (filter.subFilters && filter.subFilters.length) {
```
Comment (justinmckenzie) [inline src/components/search-filters/mobile/filters.tsx:41]: same here

File: src/hooks/search/index.ts [hunk 4]
```diff
@@ -130,13 +130,13 @@ const useSearch = (
...
                         l1.name === item ||
-                        l1.subItems.find(l2 => l2.name === item)
+                        l1.subFilters?.find(l2 => l2.name === item)
                 );
                 if (filterProduct) {
                     if (filterProduct.name !== item) {
                         // This means it's an L2 match.
                         filtersList.push(
-                            filterProduct.subItems.find(
+                            filterProduct.subFilters?.find(
```
Comment (justinmckenzie) [inline src/hooks/search/index.ts:139]: Guess it technically doesn't matter, but wouldn't you want to chain it on the first one because you might still run into the chance of `subFilters` being undefined, `filterProduct?.subFilters.find(...)`

File: src/hooks/search/utils.ts [hunk 5]
```diff
@@ -173,8 +184,8 @@ export const getFilters = async (
                     const filterItem: FilterItem = {
                         type: tag.type,
                         name: tag.name,
-                        count: allFilters || type === contentType ? 1 : 0,
-                        subItems: [],
+                        // count: allFilters || type === contentType ? 1 : 0,
```
Comment (justinmckenzie) [inline src/hooks/search/utils.ts:187]: comment

File: src/hooks/search/utils.ts [hunk 6]
```diff
@@ -185,37 +196,42 @@ export const getFilters = async (
...
+            item.subFilters.sort(
+                (prev, next) => (next.count || 0) - (prev.count || 0)
+            );
         }
     });
 
     const l1Items = filterItems
         .filter(({ type }) => type === 'L1Product')
-        .sort((prev, next) => next.count - prev.count);
+        .sort(sortFunction);
```
Comment (justinmckenzie) [inline src/hooks/search/utils.ts:212]: 👍

**Review by justinmckenzie — APPROVED** [review]
optional code cleanup, lgtm

**Review by ajhenri — APPROVED** [review]
Looks great. Very cool that we have our own component package now.

File: src/hooks/search/index.ts (hunk 4 above)
Comment (mcolella14) [reply src/hooks/search/index.ts:139]: So if I'm [reading this right](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining), I think it should still be `subFilters?.` because the optional chain checks to see if the property with the `?` appended is undefined before trying to access a further property of it. `filterProduct` will always be defined, but `filterProduct.subFilters` might not be so we add the `?` to `subFilters`?

Comment (justinmckenzie) [reply src/hooks/search/index.ts:139]: oh word okay my mistake then, I always figured you would want to first check the property existing and if it exists, then you can safely call the method on it. Thanks for clarifying and ignore any other comments of mine lol

Comment (mcolella14) [reply src/hooks/search/index.ts:139]: lol all good, i feel like they could have done a better job making optional chaining more intuitive, I'm always looking at that page for reference.
